### PR TITLE
Feature/mat 3905 cql library model

### PR DESCRIPTION
**Pull request by nmorasb**
## MADiE PR

Jira Ticket: [MAT-3905](https://jira.cms.gov/browse/MAT-3905)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is in to the **correct branch**.
* [x] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [x] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependancies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

**Review by gregory-akins — APPROVED** [review]

File: src/components/createNewCqlLibrary/CreateNewCqlLibrary.test.tsx [hunk 1]
```diff
@@ -118,6 +124,82 @@ describe("Create New Cql Library Component", () => {
...
+  //   expect(input.value).toBe("TestingLibraryName12");
+  //   const modelDropdown = screen.getByRole("button", {
+  //     name: /select a model/i,
+  //   });
+  //   expect(modelDropdown).toBeInTheDocument();
+  //   screen.debug();
+  //   const saveButton = screen.getByTestId("create-new-cql-library-save-button");
+  //   await waitFor(() => {
+  //     expect(saveButton).toBeDisabled();
+  //   });
```
Comment (RohitKandimalla) [inline src/components/createNewCqlLibrary/CreateNewCqlLibrary.test.tsx:200]: Why is this test case commented ? I think its good to have, I missed it initially. good catch.

File: src/components/createNewCqlLibrary/CreateNewCqlLibrary.test.tsx [hunk 2]
```diff
@@ -187,6 +278,15 @@ describe("Create New Cql Library Component", () => {
...
     expect(input.value).toBe("TestingLibraryName12");
+    const modelDropdown = screen.getByRole("button", {
+      name: /select a model/i,
+    });
+    userEvent.click(modelDropdown);
+    const qiCoreOption = screen.getByText("QI-Core");
+    expect(qiCoreOption).toBeInTheDocument();
+    userEvent.click(qiCoreOption);
+    const qiCore = await screen.findByText("QI-Core");
+    expect(qiCore).toBeInTheDocument();
```
Comment (RohitKandimalla) [inline src/components/createNewCqlLibrary/CreateNewCqlLibrary.test.tsx:289]: May be we can reduce constants by adding await inside the expect itself.

File: src/components/createNewCqlLibrary/CreateNewCqlLibrary.test.tsx [hunk 3]
```diff
@@ -118,6 +124,82 @@ describe("Create New Cql Library Component", () => {
...
+    expect(qiCoreOption).toBeInTheDocument();
+    userEvent.type(modelDropdown, "{esc}");
+    userEvent.dblClick(input);
+    await waitFor(
+      () => {
+        expect(
+          screen.getByText("A CQL library model is required.")
+        ).toBeInTheDocument();
+      },
+      { timeout: 5000 }
```
Comment (RohitKandimalla) [inline src/components/createNewCqlLibrary/CreateNewCqlLibrary.test.tsx:175]: Do we need a timeout, i believed waitFor can handle that.

File: src/components/createNewCqlLibrary/CreateNewCqlLibrary.test.tsx (hunk 2 above)
Comment (nmorasb) [reply src/components/createNewCqlLibrary/CreateNewCqlLibrary.test.tsx:289]: we could, I think I was using more verbose language to try to improve readability. I was also trying to follow guides recommended by [Kent C Dodds here](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library) and at first he starts out with them separate but then switches to a mix of defining the constants on one line and performing assertions on the next line, and doing it all on one line.
So, I don't know really haha

File: src/components/createNewCqlLibrary/CreateNewCqlLibrary.test.tsx (hunk 3 above)
Comment (nmorasb) [reply src/components/createNewCqlLibrary/CreateNewCqlLibrary.test.tsx:175]: I'll double check. I think at one point I was having trouble with the waitfor's default timeout not being enough but I'll test this again and see if I can remove it.

File: src/components/createNewCqlLibrary/CreateNewCqlLibrary.test.tsx (hunk 1 above)
Comment (nmorasb) [reply src/components/createNewCqlLibrary/CreateNewCqlLibrary.test.tsx:200]: I ended up just including this assertion in the test above

Comment (nmorasb) [reply src/components/createNewCqlLibrary/CreateNewCqlLibrary.test.tsx]: I added another test (which took forever cause I needed a waitfor for some reason to get past one expect) to take the place of this

File: src/components/createNewCqlLibrary/CreateNewCqlLibrary.test.tsx (hunk 3 above)
Comment (nmorasb) [reply src/components/createNewCqlLibrary/CreateNewCqlLibrary.test.tsx]: this was removed

File: src/components/createNewCqlLibrary/CreateNewCqlLibrary.test.tsx (hunk 2 above)
Comment (nmorasb) [reply src/components/createNewCqlLibrary/CreateNewCqlLibrary.test.tsx]: I renamed this variable to provide clarity over what makes it different from the previous search and click

Comment (RohitKandimalla) [reply src/components/createNewCqlLibrary/CreateNewCqlLibrary.test.tsx]: make sense, Good to know. very Interesting things in that article. I believe I did used some of the items he mentioned not to use. haha !!

**Review by RohitKandimalla — APPROVED** [review]

**Review by adongare — APPROVED** [review]
